README.txt
####################
### WATER EMBLEM ###
####################

First off, PLACEHOLDER NAME (probably). \
Second, this is a FANGAME, or derivative work, of Kantai Collection, or KanColle, by DMM and Kadokawa Games.

#############
## What is ##
#############

The concept of this project is to create a game reminescent of tactical RPGs much like Fire Emblem or Advance Wars. Mechanic-wise, the gameplay will likely be much closer to the former than latter.

This game will take many mechanics and features of KanColle and attempt to integrate them into this game whilst replacing the sortie mechanics with that of a tactical rpg. Effectively, try to imagine a KanColle with an actual battle mechanic, lol

#############
## Authors ##
#############

Current authors are:
Ippikiryu, or Flan
Remi_Scarlet, or Remi

#############
## Credits ##
#############

Even though we don't have jack shit yet, obvious credits go to DMM and Kadokawa Games for making KanColle to begin with. 


###############
## Licensing ##
###############

This work is attributed under the Fair Use license. This game is created as an academic exercise for the authors to apply their programming skills and knowledge while attempting to entertain fellow fans.
All materials originally from Kantai Collection are exclusively DMM and Kadokawa Games' intellectual property. All original material used in this exercise such as code and original graphics are copyrighted under Remi_Scarlet and Ippikiryu. All original material such as the code is licensed under a Creative Commons Attribution-NonCommercial 4.0 International License. (http://creativecommons.org/licenses/by-nc/4.0/)

This page is subject to change without notice.

src/changelog.txt
6/3/2014 - Remi
	- Added basic map initialization with a dummy map. 
	- Very crude and basic map drawing with no textures. (Solid colors)
	- Added area at bottom for UI. Gamescreen is now split into the game's "board" and "info" sections.
	- Some kind of base class for info UI manipulation
	- Proposal UI storyboards

6/4/2014 - Remi
	- Map scrolling
	- Terrain info based on cursor position
	- Added temp character portrait of Kaga <3
	- Fixed event handling to process diagonal/multi-key inputs
	- Minimap
	- Minimap scrolling
	- Added stats display and whatnot
	- Added displaying (horrible) sprites of kanmusu
	- Kanmusu are now movable. (But buggy)(Can stack)
	- Kanmusu don't stack by accident now wheee

6/5/2014 - Remi
	- Kanmusu's movable distance limited by speed stat.
	- When moving kanmusu, movable tiles are highlighted blue
	- When moving kanmusu, unmovable tiles are highlighted red
	- Changed how tiles are handled internally to allow versatility and modulabiilty

6/14/2014 - Remi
	- Mostly resource gathering. Added all sounds and images of all kanmusu. (500mb!)

6/15/2014 - Remi
	- Added music. Currently uses battle theme in game in-loop and some basic sound effects. Amazing how much of a different it makes.
	- Added enemy vessels. Wo, Re and Ru.

6/16/2014 - Remi
	- Added context menu. Battle function still not implemented.

6/21/2014 - Remi
	- Teehee forgot to update the changelog for the past couple of days.
	- Added battle animation (Unfinished, but mostly there)
	- Added character voice loading from sounds.py. Will only load voices of kanmusu you have in your fleet.
	- Changed how set_vol works so it just cycles through the __dict__... Not sure why I didn't think of that before.

6/23/2014 - Remi
	- Teehee forgot yesterday's changelog.
	- Fully implemented tick-based turn mechanic.
	- Added attack range.
	- Added attack range overlay
	- Polished up misc backend stuff here and there.

6/24/2014 - Remi
	- Enemies are now impassable
	- Started (?) AI work.

6/25/2014 - Remi
	- Implemented Fog of War
	- TORPEDOES CREATES A PRE DETERMINED PATH I DON'T KNOW HOW IT WORKS BUT IT WORKS

src/sounds.py
#########################################
######  Sound Initialisation File  ######
#########################################
import os
import pygame

class SFX(object):
	def __init__(self):
		self.dismantle = pygame.mixer.Sound(os.path.join('sounds','sfx','2-4-11.ogg'))
		self.cancel = pygame.mixer.Sound(os.path.join('sounds','sfx','Cancel.ogg'))
		self.select = pygame.mixer.Sound(os.path.join('sounds','sfx','Select.ogg'))
		self.battleStart = pygame.mixer.Sound(os.path.join('sounds','sfx','Battle Start.ogg'))
		self.resupply = pygame.mixer.Sound(os.path.join('sounds','sfx','Resupply.ogg'))
		self.victoryS = pygame.mixer.Sound(os.path.join('sounds','sfx','Victory S.ogg'))
		self.victoryA = pygame.mixer.Sound(os.path.join('sounds','sfx','Victory A.ogg'))
		self.victoryB = pygame.mixer.Sound(os.path.join('sounds','sfx','Victory B.ogg'))
		self.repairList = pygame.mixer.Sound(os.path.join('sounds','sfx','Repair List.ogg'))
		self.questComplete = pygame.mixer.Sound(os.path.join('sounds','sfx','Quest Complete.ogg'))
		self.equipment = pygame.mixer.Sound(os.path.join('sounds','sfx','Equipment.ogg'))
		self.craftFail = pygame.mixer.Sound(os.path.join('sounds','sfx','Craft Fail.ogg'))
		self.compass = pygame.mixer.Sound(os.path.join('sounds','sfx','Compass.ogg'))
		self.modernize = pygame.mixer.Sound(os.path.join('sounds','sfx','Modernize.ogg'))
		self.sonarShort = pygame.mixer.Sound(os.path.join('sounds','sfx','Sonar (Short).ogg'))
		self.sonarLong = pygame.mixer.Sound(os.path.join('sounds','sfx','Sonar (Long).ogg'))
		self.cursorMove = pygame.mixer.Sound(os.path.join('sounds','sfx','Cursor Move.ogg'))

	def setVolume(self, vol):
		'''self.dismantle.set_volume(vol)
		self.cancel.set_volume(vol)
		self.select.set_volume(vol)
		self.battleStart.set_volume(vol)
		self.resupply.set_volume(vol)
		self.victoryS.set_volume(vol)
		self.victoryB.set_volume(vol)
		self.victoryA.set_volume(vol)
		self.repairList.set_volume(vol)
		self.questComplete.set_volume(vol)
		self.equipment.set_volume(vol)
		self.craftFail.set_volume(vol)
		self.compass.set_volume(vol)
		self.modernize.set_volume(vol)
		self.sonarShort.set_volume(vol)
		self.sonarLong.set_volume(vol)
		self.cursorMove.set_volume(vol)'''
		for key in self.__dict__:
			eval("self."+key+".set_volume(vol)")


class BGM(object):
	def __init__(self):
		self.battleBoss = os.path.join('sounds','bgm','Battle Boss.ogg')
		self.battle = os.path.join('sounds','bgm','Battle.ogg')
		self.eventBattle = os.path.join('sounds','bgm','Event Battle.ogg')
		self.eventBattle2 = os.path.join('sounds','bgm','Event Battle 2.ogg')
		self.eventBoss = os.path.join('sounds','bgm','Event Boss.ogg')
		self.eventBoss2 = os.path.join('sounds','bgm','Event Boss 2.ogg')
		self.port = os.path.join('sounds','bgm','Port.ogg')
		self.quests = os.path.join('sounds','bgm','Quests.ogg')
		self.Sortie = os.path.join('sounds','bgm','Sortie.ogg')


#Because loading every voice for every ship in the game is a bad idea.
#Load only the ships being used. Eg, in fleet, in the battle map, etc.
#Don't load ship voices that we KNOW aren't going to be used.
#kanmusu should be supplied as a string.
class Voices(object):
	def __init__(self, kanmusu):
		print kanmusu
		self.intro = pygame.mixer.Sound(os.path.join('sounds','kanmusu',kanmusu,'1 - Intro.ogg'))
		self.secretary1 = pygame.mixer.Sound(os.path.join('sounds','kanmusu',kanmusu, '2 - Secretary1.ogg'))
		self.secretary2 = pygame.mixer.Sound(os.path.join('sounds','kanmusu',kanmusu, '3 - Secretary2.ogg'))
		self.secretary3 = pygame.mixer.Sound(os.path.join('sounds','kanmusu',kanmusu, '4 - Secretary3.ogg'))
		self.shipConstructed = pygame.mixer.Sound(os.path.join('sounds','kanmusu',kanmusu, '5 - Ship Constructed.ogg'))
		self.expeditionComplete = pygame.mixer.Sound(os.path.join('sounds','kanmusu',kanmusu, '6 - Expedition Complete.ogg'))
		self.returnFromSortie = pygame.mixer.Sound(os.path.join('sounds','kanmusu',kanmusu, '7 - Return from Sortie.ogg'))
		self.showRank = pygame.mixer.Sound(os.path.join('sounds','kanmusu',kanmusu, '8 - ShowRank.ogg'))
		self.equipment1 = pygame.mixer.Sound(os.path.join('sounds','kanmusu',kanmusu, '9 - Equipment Resupply1.ogg'))
		self.equipment2 = pygame.mixer.Sound(os.path.join('sounds','kanmusu',kanmusu, '10 - Equipment Resupply2.ogg'))
		self.dockingLight = pygame.mixer.Sound(os.path.join('sounds','kanmusu',kanmusu, '11 - Docking Light.ogg'))
		self.dockingHeavy = pygame.mixer.Sound(os.path.join('sounds','kanmusu',kanmusu, '12 - Docking Heavy.ogg'))
		self.joiningFleet = pygame.mixer.Sound(os.path.join('sounds','kanmusu',kanmusu, '13 - Joining Fleet.ogg'))
		self.startSortie = pygame.mixer.Sound(os.path.join('sounds','kanmusu',kanmusu, '14 - Start Sortie.ogg'))
		self.battleStart = pygame.mixer.Sound(os.path.join('sounds','kanmusu',kanmusu, '15 - Battle Start.ogg'))
		self.attack = pygame.mixer.Sound(os.path.join('sounds','kanmusu',kanmusu, '16 - Attack.ogg'))
		self.nightAttack = pygame.mixer.Sound(os.path.join('sounds','kanmusu',kanmusu, '17 - Night Attack.ogg'))
		self.nightBattle = pygame.mixer.Sound(os.path.join('sounds','kanmusu',kanmusu, '18 - Night Battle.ogg'))
		self.underFire1 = pygame.mixer.Sound(os.path.join('sounds','kanmusu',kanmusu, '19 - Under Fire1.ogg'))
		self.underFire2 = pygame.mixer.Sound(os.path.join('sounds','kanmusu',kanmusu, '20 - Under Fire2.ogg'))
		self.heavyDamage = pygame.mixer.Sound(os.path.join('sounds','kanmusu',kanmusu, '21 - Heavy Damage.ogg'))
		self.sink = pygame.mixer.Sound(os.path.join('sounds','kanmusu',kanmusu, '22 - Sink.ogg'))
		self.mvp = pygame.mixer.Sound(os.path.join('sounds','kanmusu',kanmusu, '23 - MVP.ogg'))
		self.libraryInfo = pygame.mixer.Sound(os.path.join('sounds','kanmusu',kanmusu, '25 - Library Info.ogg'))
		self.equipment3 = pygame.mixer.Sound(os.path.join('sounds','kanmusu',kanmusu, '26 - Equipment3.ogg'))
		self.resupply = pygame.mixer.Sound(os.path.join('sounds','kanmusu',kanmusu, '27 - Resupply.ogg'))
	def setVolume(vol):
		for key in self.__dict__:
			eval("self."+key+".set_volume(vol)")